README.md

```diff
@@ -0,0 +1,269 @@
+## Generic Bucket Provisioning
+
+Kubernetes natively supports dynamic provisioning for many types of file and
+block storage, but lacks support for object bucket provisioning. 
+This repo is a temporary placeholder for bucket provisioning CRDs and related
+generated client code. The longer term goal is to move this repo to a Kubernetes
+repo such as sig-storage or an external storage repo.
+
+### Design Overview
+S3 compatible object store buckets are stable and consistent among most, if not
+all, object store solutions. Therefore, the time has come where we can
+support a bucket provisioning API similar to that used for Persistent Volumes.
+We propose two new Custom Resources to abstract an object store bucket and
+a claim/request for such a bucket. 
+
+An `ObjectBucketClaim` (OBC) is similar in usage to a Persistent Volume Claim
+(PVC), and an `ObjectBucket` (OB) is the Persistent Volume (PV) equivalent. Only 
+dynamic bucket provisioning will be supported. In other words, a storage admin
```

Comment:
> Only dynamic bucket provisioning will be supported.

Again, we don't support ANY kind of provisioning in this design, that is left entirely up to operator authors.  The only thing this project offers is a common API that could be employed by all S3 providers.